akka-javasdk/src/test/java/akka/javasdk/agent/DummyAgent.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.javasdk.agent;
+
+class DummyAgent1 extends ChatAgent {
+  Effect<String> doSomething(String question) {
+    return effects()
+        .systemMessage("You are a helpful...")
```

Comment:
ok, how to "inject" persisted prompts? See discussion here: https://github.com/lightbend/kalix-runtime/issues/3832

Reply:
This would be the default if there is no persisted prompt for the agent name (or if we change that to component id), persisted overrides this one.

Reply:
`persisted overrides this one.`

I think I would prefer to code that explicitly. When I just want to use static, I could do as above, but for editable/persisted prompts, this could be:
```
effects()
        .systemMessage(prompts().get("my-system-prompt"))
``` 

I think we should also keep in mind that for a given agent, we could have many possible system prompts, and many possible user prompts, depending on some custom logic, I could use one or another. It would be hard to keep that under one component/agent id.

Reply:
IMO the convenient standard path shouldn't be imperative but declarative, so defaulting to agent id, but perhaps allowing to specify a separate prompt id together with a starting point prompt in code, seems good enough for me. 

A user could anyway use whatever client, including the prompt client, to call some thing to get a text back and use for prompt before they use the effect builder.

Side note, not that I'm strongly against that flexibility but, is that really very likely, that you would have several different prompts for the same agent, how is it then the same agent? Isn't it generally that for a specific agent you'd have one system prompt that you want to tweak over time, and then a user prompt template that you tweak over time and fold user input into?

Reply:
> but, is that really very likely, that you would have several different prompts for the same agent 

I have no idea :) I'm just trying to think about some edge cases. Maybe it will be easier to have 2+ different user prompts, instead of complex templating.

Reply:
I don't think it's useful to have an agent that can run with different prompts depending on conditions. Different behavior, different agents. 

If that's needs one should do an if-and-else before calling AgentA or AgentB.

Reply:
This might not be a completely different prompt, but a prompt that is required for that behavior. For instance, I have an agent that will respond with a news brief from the last day. We are able to detect some information about the user who calls this agent, so my system prompt could be:
```
if (user.age < 17) return effect().systemMessage("act as teenager that is not that interested into politics")...
else if (user.age >= 18 && user.age <= 60) return effect().systemMessage("act as adult that ...")
else if (user.age >=  60)  return effect().systemMessage("act as adult that don't care about the latest tech news...")
```

Reply:
I agree, such logic should be possible inside the agent, and I see no problem supporting it

Reply:
Circling that back to the discussion on `reply()` and `error()`. 

We could also think of: `if (user.age < 18) return effect().error(..)`.

I think the bottom line is, the handler is a method like any other that one can use to write logic and therefore switching prompt, returns error, enriching user messages, filling in templates using your own template engine. That's all should be possible.